File: python/models/claude_3.py
import boto3
import json

client = boto3.client(service_name="bedrock-runtime", region_name="us-east-1")

prompt = "Hello, how are you today?"
model_id = "anthropic.claude-3-haiku-20240307-v1:0"

request_body = json.dumps({
    "anthropic_version": "bedrock-2023-05-31",
    "max_tokens": 1000,
    "messages": [{
        "role": "user",
        "content": [{"type": "text", "text": prompt}]
    }
    ]
})

response = client.invoke_model(
    modelId=model_id,
    body=request_body
)

result = json.loads(response.get("body").read())
message = result["content"][0]["text"]

print(f"Response: {message}")

File: javascript/models/claude_3.js
import {BedrockRuntimeClient, InvokeModelCommand} from "@aws-sdk/client-bedrock-runtime";

const client = new BedrockRuntimeClient({region: "us-east-1"});

const prompt = "Hello, how are you today?";
const modelId = "anthropic.claude-3-haiku-20240307-v1:0";

const body = JSON.stringify({
    anthropic_version: "bedrock-2023-05-31",
    max_tokens: 1000,
    messages: [{
        role: "user",
        content: [{type: "text", text: prompt}]
    }]
});

const response = await client.send(new InvokeModelCommand({
    body, modelId, contentType: "application/json",
}));

const result = JSON.parse(new TextDecoder().decode(response.body));
const message = result.content[0].text;

console.log(`Response: ${message}`);
